contracts/ChainlinkRegistry/ChainlinkRegistry.sol

```diff
@@ -39,6 +38,15 @@
     return _feed;
   }
 
+  function _isProxy(address _feed) internal view returns (bool) {
+    if (_feed == address(0)) return false;
+    try IAggregatorProxy(_feed).aggregator() returns (address) {
+      return true;
+    } catch {
+      return false;
+    }
+  }
```

Comment:
I'm really wondering if this is worth having on-chain... Since we are already doing the information aggregation off-chain, why not just send `isProxy` or not? Maybe having it on-chain saves us from going one contract by one, or we can simply add this check (`.aggregator()`) on our off-chain script.
Thinking out loud here.

Reply:
Interesting idea. My initial approach was to make this on-chain so that:
1. It was simpler to execute
1. We couldn't screw up (if we set it incorrectly, it could have some bad consequences down the line

But, I do see your point in that we could save the gas cost of checking this on-chain

I'm honestly not sure what the best solution is. Now that I explained my POV, what do you think?

Reply:
I'd rather be sure we don't screw up, so let's make it on-chain. This is not going to be a cost that the user will pay. We will just pay for safety, so, its ok!